src/satosa/exception.py

```diff
@@ -30,6 +30,39 @@ class SATOSACriticalError(SATOSAError):
     """
     pass
 
+class SATOSAProcessingHaltError(SATOSAError):
+    """
+    SATOSA error that should stop processing
+    """
+    def __init__(self, state, message, *args, **kwargs):
+        """
+        :type state: satosa.state.State
+        :param message: str
+        :param args: any
+        :param kwargs: any
+
+        :param state: Satosa state
+        :param message: A test message
+        :param args: whatever
+        :param kwargs: whatever
+        """
+        super().__init__(message, *args, **kwargs)
+        self._message = "One of the plugins or microservices raised an error. Redirect to an error page"
```

Comment:
> But then, I am thinking, why shouldn't we redirect to an error page, for any error that is not handled deeper? ie, why should SATOSANoBoundEndpointErrors not also be redirected to a page where the error message is shown in a nice/user-friendly way? (This would require we build a template/html-page to display errors) 

True that ! I wouldn't have anything against that and it fits my our InAcademia use case fine. 

Something like : 

``` python
        try:
            self._load_state(context)
            spec = self.module_router.endpoint_routing(context)
            resp = self._run_bound_endpoint(context, spec)
            self._save_state(resp, context)
        except Exception as err:
            satosa_logging(logger, logging.ERROR,
                           "Request processing stopped due to an error. Redirecting to an error page..",
                           context.state,
                           exc_info=True)
            satosa_logging(logger, logging.ERROR, err.message, context.state, exc_info=True)
            if err.redirect_uri:
                resp = SeeOther(redirect_url=err.redirect_uri)
            else:
                resp = ServiceError(message=err.message)
        return resp
```

?